hack/release.md

```diff
@@ -50,36 +50,13 @@ Examples:
 
 _Note: only Knative admins can create versioned releases._
```

Comment:
Tekton admins?

Reply:
hm it probably is still actually Knative admins right now, I'm guessing that's how the permissions are set up 🤔